src/poetry/repositories/repository.py

```diff
@@ -2,18 +2,17 @@
 
 from typing import TYPE_CHECKING
 
-from poetry.repositories.base_repository import BaseRepository
+from poetry.repositories.base import BaseRepository
 
 
 if TYPE_CHECKING:
     from poetry.core.packages.dependency import Dependency
     from poetry.core.packages.package import Package
-    from poetry.core.packages.utils.link import Link
 
 
 class Repository(BaseRepository):
     def __init__(self, packages: list[Package] = None, name: str = None) -> None:
-        super().__init__()
+        super().__init__(name)
```

Comment:
```suggestion
        super().__init__(name, packages)
```

Then the rest of `__init__` can be dropped.